components/forms/CreateTxForm/index.tsx

```diff
@@ -49,10 +51,11 @@ const CreateTxForm = ({ router, senderAddress, accountOnChain }: CreateTxFormPro
     try {
       assert(typeof accountOnChain.accountNumber === "number", "accountNumber missing");
       assert(msgGetters.current.length, "form filled incorrectly");
+      assert(state.chain.chainId, "chainId missing");
 
       const msgs = msgGetters.current
-        .filter(({ isMsgValid, msg }) => isMsgValid(msg))
-        .map(({ msg }) => msg);
+        .filter(({ isMsgValid }) => isMsgValid())
+        .map(({ msg }) => exportMsgToJson(msg));
```

Comment:
Nice! Just one note: The `toJSON` method on the messages does not really return JSON. It is more a JavaScript object ready to be serialized to JSON. So this is correct in combination with the `dataJSON: JSON.stringify(tx),` below.